src/toil/worker.py

```diff
@@ -563,19 +567,68 @@ def workerScript(jobStore, config, jobName, jobStoreID, redirectOutputToLogFile=
         return 1
     else:
         return 0
+        
+def parse_args(args):
+    """
+    Parse command-line arguments to the worker.
+    """
+    
+    # Drop the progrram name
+    args = args[1:]
+    
+    # Make the parser
+    parser = argparse.ArgumentParser()
+    
+    # Now add all the options to it
+    
+    # Base required job information
+    parser.add_argument("jobName", type=str,
+        help="Text name of the job being run")
+    parser.add_argument("jobStoreLocator", type=str,
+        help="Information required to connect to the job store")
+    parser.add_argument("jobStoreID", type=str,
+        help="ID of the job within the job store")
+    
+    # Additional worker abbilities
+    parser.add_argument("--context", default=[], action="append",
```

Comment:
I'm worried that this is potentially confusing and obscure as a public facing feature.

It took me a minute to realize you're probably passing them in pickled on the commandline because that's probably the only way it would accept context manager classes(?).  Otherwise, I would say if it needs to be pickled/unpickled, the functions should do that lifting for you and just pass in classes and get out classes.  It's difficult though as this seems to break functionality from the actual command line so these args would always have to be crafted within the python API?

Should this be a public facing arg?  Even if you feel these args are internal, the help for this this should probably be expanded as the need to pass in arbitrary (pickled) context managers isn't entirely clear?  I would perhaps even include information about the use of this for cleaning up.

Reply:
This isn't really supposed to be a public facing feature, just an internal interface between the leader and the worker.

Previously the worker just parsed its arguments directly out of sys.argv. I've added argparse into the mix because I wanted to extend that protocol and didn't want to write a bunch of list slicing code by hand, and because I was tired of trying to remember which argument number was which. But nobody should actually be running `_toil_worker --help` and trying to construct arguments for it, should they?

I suppose we could turn these from pickled objects into references to files in the job store, just like the jobs themselves are, and make the worker fetch them that way. Then a human could conceivably point to one for a normal human reason.

I'll expand the help string a little bit to make it clear why we have this feature at all, since it's pretty weird.

Reply:
Alright, that looks good to me then.  Thanks for expanding the help string!